### PR TITLE
ec2instanceconnect 1.0.0 (new formula)

### PR DESCRIPTION
**Pull request by webframp**
Adds a recipe for the mssh and msftp commands from the aws ec2 instance
connect cli.

https://github.com/aws/aws-ec2-instance-connect-cli

Does not conflict with standard ssh,sftp and can be used alongside
existing installs of those tools.

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by webframp** [comment]
While I can definitely see the value of an automated check for lack of notoriety, this is still an AWS provided tool and a repository under their own organization.

File: Formula/ec2instanceconnect.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
+class Ec2instanceconnect < Formula
+  include Language::Python::Virtualenv
+
+  desc "All-in-one client for EC2 Instance Connect"
+  homepage "https://github.com/aws/aws-ec2-instance-connect-cli"
+  url "https://files.pythonhosted.org/packages/0c/7b/2dbd14b2caf0d834f781ef982bb4b22884245c44af0ebb6ccc5a7019fd3a/ec2instanceconnectcli-1.0.0.tar.gz"
+  sha256 "dc818c78d51a475a97d502db0221962e3c42af66900a63d7da69d3d68af29de8"
+  revision 1
```
Comment (chenrui333) [inline Formula/ec2instanceconnect.rb]:
```suggestion
```

File: Formula/ec2instanceconnect.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  head "https://github.com/aws/aws-ec2-instance-connect-cli.git", :branch => "master"
+
+  depends_on "python"
+
+  resource "botocore" do
+    url "https://files.pythonhosted.org/packages/56/28/9ef10dca4aa0d48802765447e3f6371dc218567b0f3038bdb7664ae4f0e0/botocore-1.12.253.tar.gz"
```
Comment (chenrui333) [inline Formula/ec2instanceconnect.rb]: You probably can bump this to the latest, `v1.13.3`?

File: Formula/ec2instanceconnect.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/ec2instanceconnect.rb]: For the brand new version, we dont need `revision` line.

**Comment by chenrui333** [comment]
After the above PR suggestions, can you also update the commit message to this format, `ec2instanceconnect 1.0.0 (new formula)`? 

Thanks a lot for your contribution!!

File: Formula/ec2instanceconnect.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  def install
+    virtualenv_install_with_resources
+
+    bin.install_symlink libexec/"bin/mssh"
+    bin.install_symlink libexec/"bin/msftp"
+  end
+
+  test do
+    system "mssh", "-h"
```
Comment (chenrui333) [inline Formula/ec2instanceconnect.rb]: You might want to have better assertions for the test step, like the expectation of the outputs.

**Review by zbeekman — CHANGES_REQUESTED** [review]
Please address inline comments.

Also, while I appreciate that this is an "official" AWS package, that doesn't necessarily mean that Homebrew should accept it. Lots of big-important-companies™️ make open source tools to interact with their ecosystems, but that doesn't mean that they can and should expend our time and efforts to maintain and publicize those resources for them.

You're welcome to create a tap to host this package, but right now this doesn't meet the requirements.

**Comment by webframp** [comment]
@zbeekman sorry, I'm unclear why this was closed? I just pushed updates to address these.

File: Formula/ec2instanceconnect.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
+class Ec2instanceconnect < Formula
+  include Language::Python::Virtualenv
+
+  desc "All-in-one client for EC2 Instance Connect"
+  homepage "https://github.com/aws/aws-ec2-instance-connect-cli"
+  url "https://files.pythonhosted.org/packages/0c/7b/2dbd14b2caf0d834f781ef982bb4b22884245c44af0ebb6ccc5a7019fd3a/ec2instanceconnectcli-1.0.0.tar.gz"
```
[the remark this answers is not in the document]
Comment (webframp) [reply Formula/ec2instanceconnect.rb]: Sorry I missed where this was documented and I counted 200+ current python formulas using a pythonhosted.org url. I can adjust

File: Formula/ec2instanceconnect.rb (hunk 2 above)
Comment (webframp) [reply Formula/ec2instanceconnect.rb]: Adjusted

File: Formula/ec2instanceconnect.rb (hunk 1 above)
Comment (webframp) [reply Formula/ec2instanceconnect.rb]: Removed

File: Formula/ec2instanceconnect.rb [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  sha256 "dc818c78d51a475a97d502db0221962e3c42af66900a63d7da69d3d68af29de8"
+  revision 1
+  head "https://github.com/aws/aws-ec2-instance-connect-cli.git", :branch => "master"
```
[the remark this answers is not in the document]
Comment (webframp) [reply Formula/ec2instanceconnect.rb]: I can remove this

**Comment by zbeekman** [comment]
I don't think this package is appropriate due to the notoriety audit. From my review comments above:

>Also, while I appreciate that this is an "official" AWS package, that doesn't necessarily mean that Homebrew should accept it. Lots of big-important-companies™️ make open source tools to interact with their ecosystems, but that doesn't mean that they can and should expend our time and efforts to maintain and publicize those resources for them.
>
>You're welcome to create a tap to host this package, but right now this doesn't meet the requirements.

File: Formula/ec2instanceconnect.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (zbeekman) [reply Formula/ec2instanceconnect.rb]: I'm an idiot on that one. Please ignore that comment. Your usage here was correct.

**Comment by zbeekman** [comment]
Sorry, but if it gains in popularity on github, then we'd be happy to reconsider. The GH popularity is a decent predictor of whether or not people will install it through homebrew. And we'd like to avoid maintaining packages that people are not using (much) via homebrew.

**Comment by webframp** [comment]
Np @zbeekman. It's a relatively new feature for ec2, so will try again if popularity grows enough.

Moved it over here for now: https://github.com/webframp/homebrew-tap/

**Pull request closed without merging**